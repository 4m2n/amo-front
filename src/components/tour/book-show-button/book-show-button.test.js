import renderer from "react-test-renderer"
import React from "react"
import BookShowButton from "./book-show-button"

describe("components :: tour :: book-show-button", () => {
  it("renders a reservation link with the price", () => {
    const show = {
      price: "35",
      buy: "my-wonderfull-buy-link",
      isUpcoming: true,
    }
    const component = renderer.create(
      <BookShowButton show={show}/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it("renders a reservation link with \"free\" label when there is a book link but no price", () => {
    const show = {
      price: undefined,
      buy: "my-wonderfull-buy-link",
      isUpcoming: true,
    }
    const component = renderer.create(
      <BookShowButton show={show}/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it("renders an \"on site booking button\" when no book link exists, but there is a price", () => {
    const show = {
      price: "35",
      buy: undefined,
      isUpcoming: true,
    }
    const component = renderer.create(
      <BookShowButton show={show}/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it("renders a \"free\" label when no price or book link exists", () => {
    const show = {
      price: undefined,
      buy: undefined,
      isUpcoming: true,
    }
    const component = renderer.create(
      <BookShowButton show={show}/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it("renders an hyphen is the show is passed", () => {
    const show = {
      isUpcoming: false,
    }
    const component = renderer.create(
      <BookShowButton show={show}/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
