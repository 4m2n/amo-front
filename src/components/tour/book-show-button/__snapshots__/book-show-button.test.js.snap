// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components :: tour :: book-show-button renders a "free" label when no price or book link exists 1`] = `
<p
  className="price"
>
  Entrée gratuite
</p>
`;

exports[`components :: tour :: book-show-button renders a reservation link with "free" label when there is a book link but no price 1`] = `
<a
  className="btn-teal"
  href="my-wonderfull-buy-link"
  rel="noreferrer"
  target="_blank"
>
  Réserver (Entrée gratuite)
</a>
`;

exports[`components :: tour :: book-show-button renders a reservation link with the price 1`] = `
<a
  className="btn-teal"
  href="my-wonderfull-buy-link"
  rel="noreferrer"
  target="_blank"
>
  Réserver (35€)
</a>
`;

exports[`components :: tour :: book-show-button renders an "on site booking button" when no book link exists, but there is a price 1`] = `
<p
  className="price"
>
  Vente sur place (35€)
</p>
`;

exports[`components :: tour :: book-show-button renders an hyphen is the show is passed 1`] = `
<p>
  -
</p>
`;
